cumulus/client/cirrus-executor/src/processor.rs

```diff
@@ -192,36 +194,46 @@ where
 		(&*self.client).import_block(block_import_params, Default::default()).await?;
 
 		let mut roots =
-			self.runtime_api.runtime_api().intermediate_roots(&BlockId::Hash(parent_hash))?;
-		roots.push(state_root.encode());
+			self.client.runtime_api().intermediate_roots(&BlockId::Hash(header_hash))?;
+
+		let state_root = state_root
+			.encode()
+			.try_into()
+			.expect("State root uses the same Block hash type which can fit into [u8; 32]; qed");
```

Comment:
https://www.ietf.org/rfc/rfc2119.txt
```suggestion
			.expect("State root uses the same Block hash type which must fit into [u8; 32]; qed");
```

Reply:
Then we need to fix those as well. It is generally a good practice to use "should" for things that are usually the way they are described and "must" when they absolutely have to be.

Reply:
I agree. Furthermore, I think it's beneficial to also introduce a code-style doc starting with fixing them since the team is growing and we're having more and more unwritten rules.